2008-06-27  Lukasz Kosewski <lkosewsk@gmail.com>

	* Remove __THROW (which expands to __attribute__ (__nothrow__) on my system from the function instantiations (as opposed to declarations) on a number of functions, as GCC > 3.2 no longer supports such a thing. 

--- argp.old/argp.h	2008-06-28 02:09:23.000000000 +0000
+++ argp/argp.h	2008-06-28 02:10:29.000000000 +0000
@@ -564,13 +564,13 @@ __argp_short_program_name(const struct a
 # endif
 
 ARGP_EI void
-__argp_usage (__const struct argp_state *__state) __THROW
+__argp_usage (__const struct argp_state *__state)
 {
   __argp_state_help (__state, stderr, ARGP_HELP_STD_USAGE);
 }
 
 ARGP_EI int
-__option_is_short (__const struct argp_option *__opt) __THROW
+__option_is_short (__const struct argp_option *__opt)
 {
   if (__opt->flags & OPTION_DOC)
     return 0;
@@ -582,7 +582,7 @@ __option_is_short (__const struct argp_o
 }
 
 ARGP_EI int
-__option_is_end (__const struct argp_option *__opt) __THROW
+__option_is_end (__const struct argp_option *__opt)
 {
   return !__opt->key && !__opt->name && !__opt->doc && !__opt->group;
 }
--- argp.old/argp-parse.c	2008-06-28 02:10:01.000000000 +0000
+++ argp/argp-parse.c	2008-06-28 02:10:46.000000000 +0000
@@ -1277,13 +1277,13 @@ weak_alias (__argp_input, _argp_input)
 /* Defined here, in case a user is not inlining the definitions in
  * argp.h */
 void
-__argp_usage (__const struct argp_state *__state) __THROW
+__argp_usage (__const struct argp_state *__state)
 {
   __argp_state_help (__state, stderr, ARGP_HELP_STD_USAGE);
 }
 
 int
-__option_is_short (__const struct argp_option *__opt) __THROW
+__option_is_short (__const struct argp_option *__opt)
 {
   if (__opt->flags & OPTION_DOC)
     return 0;
@@ -1297,7 +1297,7 @@ __option_is_short (__const struct argp_o
 }
 
 int
-__option_is_end (__const struct argp_option *__opt) __THROW
+__option_is_end (__const struct argp_option *__opt)
 {
   return !__opt->key && !__opt->name && !__opt->doc && !__opt->group;
 }
